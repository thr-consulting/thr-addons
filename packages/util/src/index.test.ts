import {ExcludeFalse, valuesToFunctions} from './index';

describe('ExcludeFalse', () => {
	it('should exclude false values', () => {
		const bool1 = true;
		const bool2 = false;
		const arr = ['string A', 555, null, undefined, bool1 && 'string B', bool2 && 'string C'];

		expect(arr.filter((Boolean as any) as ExcludeFalse)).toMatchSnapshot();
	});
});

describe('valuesToFunctions', () => {
	it('should convert values to functions', () => {
		const obj = {
			field1: 'string1',
			field2: 555,
		};
		const newObj = valuesToFunctions(obj);

		expect(newObj).toMatchSnapshot();
		expect(newObj.field1()).toEqual('string1');
		expect(newObj.field2()).toEqual(555);
	});
});
