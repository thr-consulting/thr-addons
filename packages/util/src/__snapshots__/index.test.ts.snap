// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExcludeFalse should exclude false values 1`] = `
Array [
  "string A",
  555,
  "string B",
]
`;

exports[`valuesToFunctions should convert values to functions 1`] = `
Object {
  "field1": [Function],
  "field2": [Function],
}
`;
